From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: nomorecoolnicknames <nomorecoolnicknames@users.noreply.github.com>
Date: Sat, 26 Sep 2026 02:28:43 +0300
Subject: [PATCH] apns: Russia: add T-Mobile, SberMobile, VTB, Gazprombank,
 MOTIV and Letai

These operators have their own network codes (250/26, 27, 35, 45, 50
and 62) and no entry, so mobile data only works after adding the
access point by hand. The APNs are the ones the operators publish. For
250/62 there are two: m.tinkoff, which is also in AOSP's list, and tt,
from the operator's current instructions.
---
 prebuilt/common/etc/apns-conf.xml | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/prebuilt/common/etc/apns-conf.xml b/prebuilt/common/etc/apns-conf.xml
index 21a5a7fd..1324b49f 100644
--- a/prebuilt/common/etc/apns-conf.xml
+++ b/prebuilt/common/etc/apns-conf.xml
@@ -960,14 +960,21 @@
   <apn carrier="MMS" mcc="250" mnc="17" apn="mms.usi.ru" proxy="" port="" user="" password="" mmsc="http://mms" mmsproxy="192.168.168.192" mmsport="8080" type="mms" />
   <apn carrier="Tele2 RU" mcc="250" mnc="20" apn="internet.tele2.ru" proxy="" port="" user="" password="" mmsc="" authtype="1" type="default,supl" />
   <apn carrier="Tele2 RU MMS" mcc="250" mnc="20" apn="mms.tele2.ru" proxy="" port="" user="" password="" mmsc="http://mmsc.tele2.ru" mmsproxy="193.12.40.65" mmsport="8080" authtype="1" type="mms" />
+  <apn carrier="VTB Mobile" mcc="250" mnc="26" apn="vtb" type="default,supl" />
+  <apn carrier="Letai Internet" mcc="250" mnc="27" apn="internet.letai.ru" type="default,supl" />
   <apn carrier="MegaFon Internet" mcc="250" mnc="30" apn="Internet" user="gdata" password="gdata" authtype="1" type="default,supl" />
   <apn carrier="MegaFon MMS" mcc="250" mnc="30" apn="mms" mmsproxy="10.10.10.10" mmsport="8080" mmsc="http://mmsc:8002" user="mms" password="mms" authtype="1" type="mms" />
+  <apn carrier="MOTIV Internet" mcc="250" mnc="35" apn="inet.ycc.ru" user="motiv" password="motiv" type="default,supl" />
   <apn carrier="USI GPRS" mcc="250" mnc="39" apn="internet.usi.ru" type="default,supl" />
   <apn carrier="Utel MMS" mcc="250" mnc="39" apn="mms.usi.ru" mmsproxy="192.168.168.192" mmsport="8080" mmsc="http://mms" type="mms" />
   <apn carrier="МТС-интернет" mcc="250" mnc="39" apn="internet.mts.ru" proxy="" port="" user="mts" password="mts" mmsc="" authtype="1" type="default,supl" />
   <apn carrier="МТС Центр MMS" mcc="250" mnc="39" apn="mms.mts.ru" proxy="" port="" user="mts" password="mts" mmsc="http://mmsc" mmsproxy="192.168.192.192" mmsport="8080" authtype="1" type="mms" />
   <apn carrier="Beeline internet" mcc="250" mnc="44" apn="internet.beeline.ru" proxy="" port="" user="beeline" password="beeline" mmsc="" authtype="1" type="default,supl" />
   <apn carrier="Beeline MMS" mcc="250" mnc="44" apn="mms.beeline.ru" proxy="" port="" user="beeline" password="beeline" mmsc="http://mms/" mmsproxy="192.168.94.23" mmsport="8080" authtype="1" type="mms" />
+  <apn carrier="Gazprombank Mobile" mcc="250" mnc="45" apn="gpb" type="default,supl" />
+  <apn carrier="SberMobile Internet" mcc="250" mnc="50" apn="internet.sberbank-tele.com" type="default,supl" />
+  <apn carrier="T-Mobile Internet" mcc="250" mnc="62" apn="m.tinkoff" type="default,supl" />
+  <apn carrier="T-Mobile tt" mcc="250" mnc="62" apn="tt" type="default,supl" />
   <apn carrier="МТС-интернет" mcc="250" mnc="92" apn="internet.mts.ru" proxy="" port="" user="mts" password="mts" mmsc="" authtype="1" type="default,supl" />
   <apn carrier="МТС Центр MMS" mcc="250" mnc="92" apn="mms.mts.ru" proxy="" port="" user="mts" password="mts" mmsc="http://mmsc" mmsproxy="192.168.192.192" mmsport="8080" authtype="1" type="mms" />
   <apn carrier="МТС-интернет" mcc="250" mnc="93" apn="internet.mts.ru" proxy="" port="" user="mts" password="mts" mmsc="" authtype="1" type="default,supl" />
